api/src/AppBundle/Entity/CasRec.php

```diff
@@ -366,4 +377,33 @@ public function getUpdatedAt()
     {
         return $this->updatedAt;
     }
+
+    /**
+     * @return mixed
+     */
+    public function getSource()
+    {
+        return $this->source;
+    }
+
+    /**
+     * @param mixed $source
+     * @return CasRec
+     */
+    public function setSource($source)
+    {
+        $this->source = $source;
```

Comment:
Might be worth adding a check here of `in_array($source, self::validSources())`?

Reply:
Added check, and ensured the constructor sets it via this method, instead of directly on to the `$this->source` property.